pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -457,3 +457,19 @@ def test_to_frame_datetime_tz(self):
         result = idx.to_frame()
         expected = DataFrame(idx, index=idx)
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("name", [None, "name"])
+    def test_index_map(self, name):
+        # see GH20990
+        count = 6
+        index = pd.date_range("2018-01-01", periods=count, freq="M", name=name)
+        a = pd.Series(np.arange(count), index=index)
+        a.index = a.index.map(lambda x: (x.year, x.month))
+
+        exp_index = pd.MultiIndex.from_arrays(
```

Comment:
```suggestion
        exp_index = pd.MultiIndex.from_product(((2018,), range(1, 7)), names=[name, name])
```

Would help readability